wheel/requirements.txt

```diff
@@ -0,0 +1,6 @@
+iniconfig==2.0.0
+maturin==1.4.0
+chia-blockchain==2.1.4
```

Comment:
When I `pip freeze` with this locally, I get a lot more dependencies - should we be including those in this file too, or no? I don't know much about Python dependency management

Reply:
I believe those are installed from these and that doing it this way captures those too.
Although, I also don't know much about Python dependency management so I may be wrong.

Reply:
You can think of dependency management as applying additional constraints to a system.

Compare to sudoku. You have a bunch of empty squares (the version numbers for each dependency) and a bunch of numbers that can go in those squares (in this case, the list of all versions for a particular package). Then you have to find a set of numbers that go in those squares that satisfy the constraints. A general form of this is known as a "SAT solver" https://en.wikipedia.org/wiki/SAT_solver and there is a lot of academic work in trying to get SAT solvers to be as efficient as possible because they can be applied to a lot of sudoku-like search problems.

(The comparison becomes a lot more obvious when you use `>=` constraints rather than `==` since `==` is a lot more restrictive and plain-spoken about its intention.)

Reply:
That said, there are two points I want to make:

1. I believe there's a push towards putting dependency constraints in `pyproject.toml` rather than `requirements.txt`. See for example https://github.com/Chia-Network/hsms/blob/main/pyproject.toml#L11
2. `maturin` is a build-time dependency only. It does not need to be present when you're using the compiled wheel. So its version should really go in `pyproject.toml` (which is the "this is how to build me" file); see https://github.com/Chia-Network/chia_rs/blob/main/wheel/pyproject.toml#L2